framework/components/ecl-galleries/ecl-gallery-base/ecl-gallery-base.scss

```diff
@@ -0,0 +1,96 @@
+/**
+* Gallery
+* @define gallery ; weak
+*/
+
+.ecl-gallery {
+  margin: 0;
+  padding-left: map-get($ecl-spacing, 'xs');
+  padding-right: map-get($ecl-spacing, 'xs');
+}
+
+.ecl-gallery__item {
+  overflow: hidden;
+  padding: 0;
+}
+
+// That's how we show the caption on hover
+.ecl-gallery__item:hover {
+  .ecl-gallery__caption {
+    display: block;
+  }
+}
+
+.ecl-gallery__item-container {
+  margin: map-get($ecl-spacing, 'xxxs') map-get($ecl-spacing, 'xxxs') 0;
+  width: 100%;
+
+  /**
+  * General for 2 reasons:
+  *  - ensure consistency
+  *  - allow easy integration
+  */
+  img {
+    display: block;
+    width: 100%;
+  }
+}
+
+.ecl-gallery__caption {
+  background-color: $ecl-color-shade;
+  bottom: 0;
+  color: #fff;
+  display: none;
+  font-size: map-get($ecl-font-size, 'xxs');
+  height: 4em;
+  overflow: hidden;
+  // The line is ignored because prettier wants first line to stay.
+  // Whereas the linter wants the rules to go a new line directly.
+  /* stylelint-disable-next-line */
+  padding: map-get($ecl-spacing, 'xxxs') map-get($ecl-spacing, 'xxs')
+    map-get($ecl-spacing, 'xxxs') map-get($ecl-spacing, 'xxxs');
+  position: absolute;
+  text-overflow: ellipsis;
+  width: 100%;
+}
+
+.ecl-gallery__icon-wrapper {
+  background-color: $ecl-color-shade;
+  bottom: 0;
+  color: #fff;
+  height: 2.667em;
+  overflow: hidden;
+  padding: 0;
+  position: absolute;
+  right: 0;
+  text-align: center;
+  width: 2.667em;
+
+  // Camera icon needs to be smaller in this context.
+  .ecl-icon--camera {
+    font-size: map-get($ecl-font-size, 'xxs');
+    line-height: 4em;
+  }
+}
+
+.ecl-gallery__icon {
+  font-size: map-get($ecl-font-size, 'm');
+  line-height: 2.667em;
+
+  &::before {
+    margin-right: 0;
+  }
+}
+
+@include ecl-media-breakpoint-down(sm) {
+  .ecl-gallery__item {
+    float: left;
+    width: 50%;
+  }
+}
+
+@include ecl-media-breakpoint-between(sm, md) {
```

Comment:
Why is this? When resizing the window we switch from 4 column to 1, then to 2.
To manage responsive design with the grid you should better use dedicated classes (ecl-col-sm-*, ecl-col-md-*, etc.)